src/systemctl/systemctl-start-unit.c

```diff
@@ -399,3 +399,42 @@ int verb_start(int argc, char *argv[], void *userdata) {
 
         return ret;
 }
+
+int verb_toggle(int argc, char *argv[], void *userdata) {
+        _cleanup_strv_free_ char **names = NULL;
+        sd_bus *bus;
+        int r, ret = EXIT_SUCCESS;
+
+        if (!argv[1])
+                return 0;
+
+        r = mangle_names("to toggle", strv_skip(argv, 1), &names);
+        if (r < 0)
+                return r;
+
+        r = acquire_bus(BUS_MANAGER, &bus);
+        if (r < 0)
+                return r;
+
+        STRV_FOREACH(name, names) {
+                UnitActiveState active_state;
+                bool is_active = false;
+                char *new_args[3];
+
+                r = get_state_one_unit(bus, *name, &active_state);
+                if (r < 0)
+                        return r;
```

Comment:
Doing all units one by one isn't terribly efficient, especially if the --wait option is provided. Since systemctl will wait for each unit to be fully toggled before sending the request for the next unit to be toggled.

Ideally, we'd first get the active state of each provided unit, start/stop them all at once and only then wait for all the jobs to complete.

Reply:
What if the units provided are a mix of stopped and active?
Would you run 2 jobs, one for stop and one for start?